internal/operands/tekton-tasks/tekton-tasks_test.go

```diff
@@ -0,0 +1,412 @@
+package tekton_tasks
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	tekton "github.com/kubevirt/tekton-tasks-operator/api/v1alpha1"
+	"github.com/kubevirt/tekton-tasks-operator/pkg/common"
+	tektonbundle "github.com/kubevirt/tekton-tasks-operator/pkg/tekton-bundle"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	namespace = "kubevirt"
+	name      = "test-tekton"
+)
+
+var _ = Describe("environments", func() {
+	var tt *tektonTasks
+	var mockedRequest *common.Request
+	BeforeEach(func() {
+		tt = getMockedTektonTasksOperand()
+		mockedRequest = getMockedRequest()
+	})
+
+	It("New function should return object with correct tasks", func() {
```

Comment:
Can you write a similar test to check that only allowed resources are deployed, but without accessing internal struct fields of the operand?

Reply:
I've added similar tests to other operands:

1. `It("Should create tekton-tasks resources", func() { ... })`
2. `It("should remove tekton-tasks resources on cleanup", func() { ... })`

Done.